src/profiling/meter_manager.cpp

```diff
@@ -9,22 +9,23 @@ meter_manager::meter_manager() {
     meters.emplace_back(new time_meter());
 
     // add memory consumption meter
-    // TODO
+    if (has_memory_metering) {
+        meters.emplace_back(new memory_meter());
+    }
 
```

Comment:
If the `meter_manager` constructor has responsibility for setting up the `meters` vector, and these `meters` are not meant for public consumption, then `meters` should be private.